include/boost/json/impl/monotonic_resource.ipp

```diff
@@ -18,38 +19,55 @@ namespace json {
 struct monotonic_resource::block
 {
     std::size_t const size;
+    block* next = nullptr;
     std::uintptr_t top;
-    block* next;
 
     block(
         std::size_t size_,
         block* next_)
         : size(size_)
-        , top(reinterpret_cast<
-            std::uintptr_t>(this+1))
         , next(next_)
+        , top(base_ptr())
     {
     }
 
-    void*
-    alloc(
-        std::size_t n,
-        std::size_t align) noexcept
+    std::uintptr_t
+    base_ptr() const
```

Comment:
or `data() noexcept const`

Reply:
Is there a point to doing this? It's implementation detail. `data()` usually implies that you get a pointer; this returns an integer.

Reply:
oh.... true. You called it `base_ptr` though :) maybe `base()`?

Reply:
Affirm